pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1777,6 +1777,7 @@ func (cf *changeFrontier) checkpointJobProgress(
 	}
 
 	cf.localState.SetHighwater(frontier)
+	// TODO(#137692)
```

Comment:
so right now we're writing the new checkpoint .. to the local state only. which takes effect during transient errors or during node shutdown (not implemented yet). is that right?

Reply:
I still don't fully understand how this flow works. This PR is not touching this part of the flow and we are relying on [#137692](https://github.com/cockroachdb/cockroach/issues/137692#issuecomment-2637018153) to figure out how fine grained checkpointing work would work with `SetCheckpoint`. 

I think you are right though that frontier only saves to local state during transient errors but not node shutdown. But aggregators save their aggregator frontier during node shutdown https://github.com/cockroachdb/cockroach/blob/7c3fe6a353ea741161960ff6ebc98459a86246df/pkg/ccl/changefeedccl/changefeed_processors.go#L793-L799.